lib/sww.js

```diff
@@ -52,13 +53,28 @@ ServiceWorkerWare.prototype.handleEvent = function sww_handleEvent(evt) {
 
 ServiceWorkerWare.prototype.onFetch = function sww_onFetch(evt) {
   var steps = this.router.match(evt.request.method, evt.request.url);
+
+  // Push the fallback middleware at the end of the list if it's defined
+  // also being sure that wont be executed if there is a response already.
+  if (this.fallbackMw) {
+    var fallback = this.fallbackMw;
+    steps.push(function(req, res) {
```

Comment:
After some meditation, I think you're forcing this because you want to hit it if it did not hit any other middleware as the bug shows in the title. There is another problem then. To hit the middleware is to call it (a logger mw could simply inspect the request and do nothing with it but update a DB). Right now as soon as there is a middleware, [we pass through it](https://github.com/gaia-components/serviceworkerware/blob/bug-1165864/dist/sww.js#L260). So checking if we hit something is simply to check if `steps.length > 0`.

I think this check is closer to the semantics you want for this bug but then, I foresee another problem.

Suppose we have some way to transform the request. Then I want to pass all the request to a domain via a CORS proxy so I'm tweaking the request to transform the url. Of course this middleware will be hit but I really want the default middleware to hit as well.

I would simplify the semantics for default middleware as that middleware hit in the last term. If you want to simply specify a worker with one middleware, make `use` chainable returning this and use:

```js
var sww = new ServiceWorkerWare().use(myLonelyMiddleware);
```

What do you think? Ping me if you have doubts.